### PR TITLE
feat: handle T&C photos and UI photos roll

### PR DESCRIPTION
**Pull request by josepfo**

### CODE REVIEW

**Comment by josepfo** [comment]
[Automatically capture customer photo upon acceptance of Terms & Conditions](https://trello.com/c/pjmajldc/890-automatically-capture-customer-photo-upon-acceptance-of-terms-conditions)

File: lib/customers.js [hunk 1]
```diff
@@ -625,6 +625,57 @@ function updateIdCardData (patch, id) {
     })
 }
 
+/**
+ * @param {String} imageData customer t&c photo data
+ * @returns {Promise<Object>} new patch to be applied
+ */
+function updateTxCustomerPhoto (imageData) {
```
Comment (chaotixkilla) [inline lib/customers.js:661]: This function has a lot in common with `updatePhotoCard()`. Any chance we can get a refactor going on here?

File: lib/new-admin/services/transactions.js [hunk 2]
```diff
@@ -218,4 +222,15 @@ function getTxAssociatedData (txId, txClass) {
...
+  const cashInSql = 'UPDATE cash_in_txs SET tx_customer_photo_at = $1, tx_customer_photo_path = $2 WHERE customer_id=$3 AND id=$4'
+
+  const cashOutSql = 'UPDATE cash_out_txs SET tx_customer_photo_at = $1, tx_customer_photo_path = $2 WHERE customer_id=$3 AND id=$4'
+
+  return direction === 'cashIn'
+    ? db.oneOrNone(cashInSql, [formattedData.tx_customer_photo_at, formattedData.tx_customer_photo_path, customerId, txId])
+    : db.oneOrNone(cashOutSql, [formattedData.tx_customer_photo_at, formattedData.tx_customer_photo_path, customerId, txId])
+}
+
+module.exports = { batch, single, cancel, getCustomerTransactionsBatch, getTx, getTxAssociatedData, updateTxCustomerPhoto }
```
Comment (chaotixkilla) [inline lib/new-admin/services/transactions.js]:
```suggestion
module.exports = {
  batch,
  single,
  cancel,
  getCustomerTransactionsBatch,
  getTx,
  getTxAssociatedData,
  updateTxCustomerPhoto
}
```

File: lib/routes/customerRoutes.js [hunk 3]
```diff
@@ -99,10 +100,27 @@ function triggerSuspend (req, res, next) {
...
 
+function updateTxCustomerPhoto (req, res, next) {
+  const customerId = req.params.id
+  const txId = req.params.txId
+  const tcPhotoData = req.body.tcPhotoData
+  const direction = req.body.direction
+
+  Promise.all([customers.getById(customerId), txs.getTx(txId, direction)])
+    .then(([customer, tx]) => {
+      if (!customer || !tx) { throw httpError('Not Found', 404) }
```
Comment (chaotixkilla) [inline lib/routes/customerRoutes.js]: Are we sure we want to throw a `404`? Doesn't seem right. Worst case it'd be better to throw something as a Bad Request - 400 or a custom code if we want the machine to handle this error in some way

File: migrations/1627563019030-add-customer-tc-photo-path.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const db = require('./db')
+
+exports.up = function (next) {
+  const sql = [
+    'ALTER TABLE cash_in_txs ADD COLUMN tx_customer_photo_at timestamptz, ADD COLUMN tx_customer_photo_path text',
+    'ALTER TABLE cash_out_txs ADD COLUMN tx_customer_photo_at timestamptz, ADD COLUMN tx_customer_photo_path text'
```
Comment (chaotixkilla) [inline migrations/1627563019030-add-customer-tc-photo-path.js]: `timestamptz` and `text` can be uppercased

File: new-lamassu-admin/src/pages/Customers/components/PhotosCard.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+}
+
+const PhotosCard = memo(({ frontCameraData, txPhotosData }) => {
+  const classes = useStyles()
+
+  const [photosDialog, setPhotosDialog] = useState(false)
+
+  const mapKeys = pair => {
+    const key = R.head(pair)
+    const value = R.last(pair)
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/pages/Customers/components/PhotosCard.js]: Can't this just use a regular JS deconstruction? 
```suggestion
    const [key, value] = pair
```

File: new-lamassu-admin/src/pages/Customers/components/PhotosCard.js [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+
+  const photosData = filterByPhotoAvailable(
+    addPhotoDir(standardizeKeys(R.append(frontCameraData, txPhotosData)))
+  )
+
+  const singlePhoto = R.head(photosData)
+
+  const isPhotoRollAvailable = () => {
+    return !singlePhoto
+  }
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/pages/Customers/components/PhotosCard.js]:
```suggestion
  const isPhotoRollAvailable = () => !singlePhoto
```

File: new-lamassu-admin/src/pages/Customers/components/PhotosCard.js [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+  return (
+    <>
+      <Paper className={classes.photo} elevation={0}>
+        <ButtonBase
+          disabled={isPhotoRollAvailable()}
+          className={classes.button}
+          onClick={() => {
+            setPhotosDialog(true)
+          }}>
+          {singlePhoto ? (
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/pages/Customers/components/PhotosCard.js:70]: I'm not a fan of having two different boolean uses tied to the same variable. Perhaps change line 70 to `disabled={!singlePhoto}` and get rid of `isPhotoRollAvailable()`?

File: new-lamassu-admin/src/pages/Customers/components/PhotosCard.js [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+            <div className={classes.imgWrapper}>
+              <img
+                className={classes.imgInner}
+                src={`${URI}/${item?.photoDir}/${item?.path}`}
+                alt=""
+              />
+            </div>
+          </div>
+        ))}
+      </Carousel>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/pages/Customers/components/PhotosCard.js]: Could this be moved into a separate component for general use in other parts of the system (apart from line 109)? Should be a simple, but beneficial refactoring

**Review by chaotixkilla — CHANGES_REQUESTED** [review]

File: lib/customers.js (hunk 1 above)
Comment (josepfo) [reply lib/customers.js:661]: Hey! I believe we will soon refactor this part of the code to improve folder management, etc.  But I'll ask Rafael

File: new-lamassu-admin/src/styling/icons/arrow/carousel-left-arrow.svg [hunk 9]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="13px" height="33px" viewBox="0 0 13 33" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>Simple Arrow White</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/arrow/carousel-left-arrow.svg]: This `title` node makes its content show up during mouseover icon. I believe you can delete it to keep the behavior similar to the other icons

File: new-lamassu-admin/src/styling/icons/arrow/carousel-right-arrow.svg [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="15px" height="34px" viewBox="0 0 15 34" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>Group 2 Copy</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/arrow/carousel-right-arrow.svg]: Same here

**Review by chaotixkilla — CHANGES_REQUESTED** [review]

**Review by chaotixkilla — APPROVED** [review]

**Review by RafaelTaranto — APPROVED** [review]

**Pull request merged**